processing/src/main/java/org/apache/druid/segment/column/StringDictionaryEncodedColumn.java

```diff
@@ -121,6 +121,14 @@ public HistoricalDimensionSelector makeDimensionSelector(
       @Override
       public int getValueCardinality()
       {
+        /*
+         This is technically wrong if
```

Comment:
:+1:, if we ever get around to removing group by v1 then maybe we can actually make this change

Reply:
growable off heap aggregator  ( https://github.com/apache/incubator-druid/issues/8126 ) is only real way to get rid of groupBy-v1 eventually. I am hoping more people could take a look at approach chosen and acknowledge before I spend significant time finishing the PR there.